engineapi/backup_monitor.go

```diff
@@ -77,26 +102,91 @@ func NewBackupMonitor(logger logrus.FieldLogger,
 }
 
 func (m *BackupMonitor) monitorBackups() {
+	fallbackToExponentialBackoff := false
+
+	// Periodically liner timer
 	wait.PollUntil(BackupMonitorSyncPeriod, func() (done bool, err error) {
-		if err := m.syncBackups(); err != nil {
-			m.logger.Errorf("Stop monitoring because of %v", err)
+		fallback, err := m.syncBackups()
+		if err != nil && !fallback {
+			m.logger.Errorf("Stop monitoring because of %v in linear timer", err)
 			m.Close()
-			return false, err
+			return false, nil
+		}
+		if fallback {
+			fallbackToExponentialBackoff = fallback
+			return true, nil
 		}
 		return false, nil
 	}, m.ctx.Done())
+
+	if !fallbackToExponentialBackoff {
+		return
+	}
+
+	// Exponential back off timer
+	m.logger.Info("Fallback to exponential back off timer")
+	var (
+		initBackoff   = BackupMonitorSyncPeriod
+		maxBackoff    = 10 * time.Minute
+		resetDuration = BackupMonitorMaxRetryPeriod
+		backoffFactor = 2.0
+		jitter        = 0.0
+		clock         = clock.RealClock{}
+		retryCount    = 0
+	)
+	backoffMgr := wait.NewExponentialBackoffManager(initBackoff, maxBackoff, resetDuration, backoffFactor, jitter, clock)
+	defer backoffMgr.Backoff().Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), BackupMonitorMaxRetryPeriod)
```

Comment:
Should be able to use the monitors context as a parent, so that when the monitors context gets cancelled the child ctx can also be cancelled. Unless we need to differentiate between the 2 cases? 

Looks like we differentiate below as part of the case. So we cannot rely on the parent ctx cancellation propagation.

Reply:
Oh OK. We do differentiate the handling for these 2 ctx.